From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Sat, 4 Jul 2020 03:04:38 +0300
Subject: dt/dtoverlays: imx477: Add parameter to set camera module rotation

Add a rotation DT overlay parameter to allow specifying the camera
module mounting rotation. Set the default rotation to 180 as the module
is typically mounted upside-down.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
---
 arch/arm/boot/dts/overlays/README             | 5 +++--
 arch/arm/boot/dts/overlays/imx477-overlay.dts | 6 ++++++
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 61fe86817292..69892d01faae 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -1458,8 +1458,9 @@ Name:   imx477
 Info:   Sony IMX477 camera module.
         Uses Unicam 1, which is the standard camera connector on most Pi
         variants.
-Load:   dtoverlay=imx477
-Params: <None>
+Load:   dtoverlay=imx477,<param>=<val>
+Params: rotation                Mounting rotation of the camera sensor (0 or
+                                180, default 180)
 
 
 Name:   iqaudio-codec
diff --git a/arch/arm/boot/dts/overlays/imx477-overlay.dts b/arch/arm/boot/dts/overlays/imx477-overlay.dts
index 73d4f79d54e6..1a97eaaf4c82 100644
--- a/arch/arm/boot/dts/overlays/imx477-overlay.dts
+++ b/arch/arm/boot/dts/overlays/imx477-overlay.dts
@@ -27,6 +27,8 @@
 				VDIG-supply = <&imx477_vdig>;	/* 1.05v */
 				VDDL-supply = <&imx477_vddl>;	/* 1.8v */
 
+				rotation = <180>;
+
 				port {
 					imx477_0: endpoint {
 						remote-endpoint = <&csi1_ep>;
@@ -110,4 +112,8 @@
 			cam0-pwdn      = <&imx477_vana>,"gpio:4";
 		};
 	};
+
+	__overrides__ {
+		rotation = <&imx477>,"rotation:0";
+	};
 };
